tests/e2e/e2e_test.go

```diff
@@ -142,39 +142,40 @@ var _ = ginkgo.BeforeSuite(func() {
 	utils.SetSkipNetworkRunnerShutdown(skipNetworkRunnerShutdown)
 	utils.SetClient(runnerCli)
 
-	ginkgo.By("calling start API via network runner with skipNetworkRunnerStart "+fmt.Sprint(skipNetworkRunnerStart), func() {
-		if skipNetworkRunnerStart {
-			utils.Outf("{{green}}skipped 'start'{{/}}\n")
-		} else {
-			utils.Outf("{{green}}sending 'start' with binary path:{{/}} %q\n", utils.GetExecPath())
-			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
-			resp, err := runnerCli.Start(
-				ctx,
-				utils.GetExecPath(),
-				runner_sdk.WithPluginDir(utils.GetPluginDir()),
-				runner_sdk.WithGlobalNodeConfig(fmt.Sprintf(`{"log-level":"%s"}`, avalanchegoLogLevel)),
-				runner_sdk.WithNumNodes(5),
-				runner_sdk.WithBlockchainSpecs(
-					[]*runner_sdk_rpcpb.BlockchainSpec{
-						{
-							VmName:  vmName,
-							Genesis: utils.GetVmGenesisPath(),
-						},
-					},
-				))
-			cancel()
-			gomega.Expect(err).Should(gomega.BeNil())
-			utils.Outf("{{green}}successfully started:{{/}} %+v\n", resp.ClusterInfo.NodeNames)
-		}
-	})
+	os.Setenv("AVALANCHEGO_PATH", avalanchegoExecPath)
+
+	if skipNetworkRunnerStart {
+		utils.Outf("{{green}}skipped 'start'{{/}}\n")
+		return
+	}
```

Comment:
can we separate out the `BeforeSuite` function into smaller functions to make it easier to follow?

Prior to this, it was starting the network runner and then also spinning up a subnet, but since each test should be responsible for launching its own subnet, we should separate these into smaller functions.

Additionally, previously this would check the health of the network runner server itself, so I think if `skipNetworkRunnerStart` is true, we should still check to confirm that the network runner server is actually health and accessible since we assume that if we're not starting it, then it's because it should already be running in the background if I understand the flow correctly.

cc @gyuho to confirm

Reply:
Yes, we can separate this into smaller ones for better readability. The global variables are set before `BeforeSuite` and those variables can be used as parameters to the smaller functions.

Reply:
Sure